File: CodeForces/1901-2000/1915/B/1915B.go
// Problem 1915B: Not Quite Latin Square
// https://codeforces.com/contest/1915/problem/B
// Executed with: go version go1.19.8 linux/amd64
// Linux version: Linux kernel 6.1.0-10-amd64
// Submitted on: October 23rd, 2025
// Codeforces language used: Go 1.22.2

package main

import (
	"fmt"
	"strings"
)

func main() {
	var runs int
	fmt.Scan(&runs)
	runs *= 3

	for i := 0; i < runs; i++ {
		var input string
		fmt.Scan(&input)
		if strings.Contains(input, "?") {
			if strings.Contains(input, "A") {
				if strings.Contains(input, "B") {
					fmt.Println("C")
				} else {
					fmt.Println("B")
				}
			} else {
				fmt.Println("A")
			}
		}
	}
}

File: CodeForces/1901-2000/1915/B/1915B.php
<?php

// Problem 1915B: Not Quite Latin Square
// https://codeforces.com/contest/1915/problem/B
// Executed with: PHP 8.4.11 (cli) (built: Aug 13 2025 01:43:48) (NTS)
// Linux version: Linux kernel 6.17.0-5-generic
// Written on: October 27th, 2025
// Codeforces language used: Cannot be submitted

$runs = intval(trim(fgets(STDIN)));
$runs *= 3;

for ($run = 0; $run < $runs; $run++) {
    $string_input = trim(fgets(STDIN));
    if (str_contains($string_input, "?")) {
        if (!str_contains($string_input, "A")) {
            echo "A" . "\n";
        } else if (!str_contains($string_input, "B")) {
            echo "B" . "\n";
        } else {
            echo "C" . "\n";
        }
    }
}

?>
